src/publish/publishToRegistry.ts

```diff
@@ -11,6 +11,8 @@ import { shouldPublishPackage } from './shouldPublishPackage';
 import { validatePackageDependencies } from './validatePackageDependencies';
 import { performPublishOverrides } from './performPublishOverrides';
 
+type Unpromisify<T> = T extends Promise<infer U> ? U : never;
```

Comment:
note, this could be changed to `Awaited` at some point once we update to TS where that's available (I forget which version)